projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -401,11 +401,21 @@ private static function enqueue_contact_forms_style_script() {
 
 		wp_enqueue_script(
 			'contact-form-styles',
-			plugins_url( 'js/form-styles.js', __FILE__ ),
+			plugins_url( 'extensions/blocks/contact-form/util/form-styles.js', JETPACK__PLUGIN_FILE ),
```

Comment:
Hey @Automattic/jetpack-crew, could you help me here?

I've created this file: `projects/plugins/jetpack/extensions/blocks/contact-form/util/form-styles.js`, which I want to use in the blocks and in the Frontend.

I'm trying to enqueue this script here, but it's not working. 
My guess is that it is because `extensions/blocks/contact-form/*` isn't available in production builds. Am I right?
What would be the correct way to share (and enqueue) a script between the blocks (editor) and the Frontend?

Reply:
> My guess is that it is because extensions/blocks/contact-form/* isn't available in production builds.

That's correct; the .js files in the `extensions` directory all get transpiled and bundled into the block JS bundle that lives in `_inc/blocks` in production. They do not get shipped separately.

> What would be the correct way to share (and enqueue) a script between the blocks (editor) and the Frontend?

For other blocks, we rely on our existing `load_assets_as_required` method to load a `view.js` file on the frontend. In that file, we can include the `form-styles.js` file.
To load that file in the editor as well, we can import it in the `edit.js` file.

Something like this should help:

```diff
diff --git a/projects/plugins/jetpack/extensions/blocks/contact-form/edit.js b/projects/plugins/jetpack/extensions/blocks/contact-form/edit.js
index b7aaa6bd10..21ba32e147 100644
--- a/projects/plugins/jetpack/extensions/blocks/contact-form/edit.js
+++ b/projects/plugins/jetpack/extensions/blocks/contact-form/edit.js
@@ -33,6 +33,7 @@ import SalesforceLeadFormSettings, {
 } from './components/jetpack-salesforce-lead-form/jetpack-salesforce-lead-form-settings';
 import { withStyleVariables } from './util/with-style-variables';
 import defaultVariations from './variations';
+import './util/form-styles.js';
 
 const ALLOWED_BLOCKS = [
 	...map( childBlocks, block => `jetpack/${ block.name }` ),
diff --git a/projects/plugins/jetpack/extensions/blocks/contact-form/util/with-style-variables.js b/projects/plugins/jetpack/extensions/blocks/contact-form/util/with-style-variables.js
index 42862ee099..e9ec4d5644 100644
--- a/projects/plugins/jetpack/extensions/blocks/contact-form/util/with-style-variables.js
+++ b/projects/plugins/jetpack/extensions/blocks/contact-form/util/with-style-variables.js
@@ -1,5 +1,3 @@
-import './form-styles.js';
-
 export const withStyleVariables = WrappedComponent => props => {
 	const EDITOR_SELECTOR = '[data-type="jetpack/contact-form"]';
 	const { generateStyleVariables } = window.jetpackForms;
diff --git a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
index 579dc3bf5e..d72c9eb309 100644
--- a/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
+++ b/projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php
@@ -399,14 +399,6 @@ class Grunion_Contact_Form_Plugin {
 	private static function enqueue_contact_forms_style_script() {
 		add_filter( 'js_do_concat', array( __CLASS__, 'disable_forms_style_script_concat' ), 10, 3 );
 
-		wp_enqueue_script(
-			'contact-form-styles',
-			plugins_url( 'extensions/blocks/contact-form/util/form-styles.js', JETPACK__PLUGIN_FILE ),
-			array(),
-			JETPACK__VERSION,
-			true
-		);
-
 		if ( ! is_admin() ) {
 			wp_enqueue_script(
 				'contact-form-styles-frontend',
@@ -537,6 +529,8 @@ class Grunion_Contact_Form_Plugin {
 			);
 		}
 
+		Jetpack_Gutenberg::load_assets_as_required( 'contact-form' );
+
 		return Grunion_Contact_Form::parse( $atts, do_blocks( $content ) );
 	}
 
```

This will work in the Jetpack plugin, but in the package we do not have access to the `Jetpack_Gutenberg` class so we'll have to find a different solution there.